extensions/blocks/story/edit.native.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Text, View } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import { Image } from '@wordpress/components';
+import { BlockMediaUpdateProgress } from '@wordpress/block-editor';
+import { __, sprintf } from '@wordpress/i18n';
+import { getProtocol } from '@wordpress/url';
+import { doAction, hasAction } from '@wordpress/hooks';
+import {
+	// TODO implement the similar / following bridge signals
+	// requestImageFailedRetryDialog,
+	// requestImageUploadCancelDialog,
+	// requestImageFullscreenPreview,
+	mediaUploadSync,
+	mediaSaveSync,
+	requestMediaFilesEditorLoad,
+} from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import StoryEditingButton from './story-editing-button';
+
+class StoryEdit extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.onEditButtonTapped = this.onEditButtonTapped.bind( this );
+
+		this.mediaUploadStateReset = this.mediaUploadStateReset.bind( this );
+		this.finishMediaUploadWithSuccess = this.finishMediaUploadWithSuccess.bind( this );
+		this.finishMediaUploadWithFailure = this.finishMediaUploadWithFailure.bind( this );
+		this.updateMediaUploadProgress = this.updateMediaUploadProgress.bind( this );
+
+		// save progress bindings
+		this.mediaSaveStateReset = this.mediaSaveStateReset.bind( this );
+		this.finishMediaSaveWithSuccess = this.finishMediaSaveWithSuccess.bind( this );
+		this.finishMediaSaveWithFailure = this.finishMediaSaveWithFailure.bind( this );
+		this.updateMediaSaveProgress = this.updateMediaSaveProgress.bind( this );
+
+		this.onStorySaveResult = this.onStorySaveResult.bind( this );
+
+		this.onMediaModelCreated = this.onMediaModelCreated.bind( this );
+
+		this.state = {
+			isUploadInProgress: false,
+			isSaveInProgress: false,
+			didUploadFail: false,
+			didSaveFail: false,
+		};
+	}
+
+	componentDidMount() {
+		const { attributes } = this.props;
+		if ( attributes.mediaFiles !== undefined ) {
+			for ( let i = 0; i < attributes.mediaFiles.length; i++ ) {
+				const protocolForUrl = getProtocol( attributes.mediaFiles[ i ].url );
+				if (
+					( attributes.mediaFiles[ i ].id && protocolForUrl !== 'http:' ) ||
+					protocolForUrl !== 'https:'
+				) {
+					mediaUploadSync();
+					mediaSaveSync();
+					return;
+				}
```

Comment:
Just to clarify, if the media file satisfies either of these conditions:
- it has a value for the `id` attribute **and** its URL protocol is not `http` (e.g. `file` or `https`)
- its URL protocol is not `https` (e.g. `file` or `http`)
upload and save the file.

Could you please help me understand what these specific checks are for?

Reply:
good catch,  I meant to say
`( attributes.mediaFiles[ i ].id && (protocolForUrl !== 'http:' ||  protocolForUrl !== 'https:')) `. Also sometimes the file can lack a file descriptor when we're still saving it (getprotocol would return `undefined`) so that's why I'm checking explicitly for http or https as to get the meaning I want  of _"there's already a remote url in place"_. Will update shortly.

Reply:
> upload and save the file.

just to clarify, it goes the other way around: the file gets saved first, and only then can be uploaded

Reply:
fixed in de0d8d5be5

Reply:
BTW ended up removing the check in `componentDidMount` given we actually are interested in updates during all stages (1. when a temporary id is created but we still have an old http url because it is about to be replaced while being saved, 2. when we have a media model created and as such have a local id with a local file path, and 3. when we finished uploading and receive a remote id assignment from the server and a remote URL to finish things with).

Done in e22b73b5d7